ui/pages/permissions-connect/snaps/snaps-connect/snaps-connect.js

```diff
@@ -57,7 +58,11 @@ export default function SnapsConnect({
   const snaps = getDedupedSnaps(request, currentPermissions);
 
   const SnapsConnectContent = () => {
-    const { hostname: trimmedOrigin } = useOriginMetadata(origin) || {};
+    let trimmedOrigin = (useOriginMetadata(origin) || {})?.hostname;
+
+    if (isSnapId(origin)) {
+      trimmedOrigin = targetSubjectMetadata.name;
```

Comment:
Hmm, should this be using the localized Snaps metadata name instead?

Reply:
Makes sense, done: [3ed97f6](https://github.com/MetaMask/metamask-extension/pull/26715/commits/3ed97f6012dd19dd40833da80444eb980477c12c)